tachyon/crypto/commitments/fri/two_adic_fri_prover.h

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2022 The Plonky3 Authors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.plonky3 and the LICENCE-APACHE.plonky3
+// file.
+
+#ifndef TACHYON_CRYPTO_COMMITMENTS_FRI_TWO_ADIC_FRI_PROVER_H_
+#define TACHYON_CRYPTO_COMMITMENTS_FRI_TWO_ADIC_FRI_PROVER_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/bits.h"
+#include "tachyon/crypto/commitments/fri/two_adic_fri_config.h"
+#include "tachyon/crypto/commitments/fri/two_adic_fri_proof.h"
+#include "tachyon/math/finite_fields/extension_field_traits_forward.h"
+#include "tachyon/math/matrix/matrix_types.h"
+#include "tachyon/zk/air/plonky3/challenger/challenger.h"
+
+namespace tachyon::crypto {
+
+template <typename ExtF, typename ChallengeMMCS, typename Challenger>
+CommitPhaseResult<ChallengeMMCS> CommitPhase(
+    TwoAdicFriConfig<ChallengeMMCS>& config,
+    std::vector<std::vector<ExtF>>&& inputs, Challenger& challenger) {
+  using Commitment = typename ChallengeMMCS::Commitment;
+  using ProverData = typename ChallengeMMCS::ProverData;
+
+  std::vector<ExtF> folded = std::move(inputs[0]);
+  size_t sz = base::bits::CheckedLog2(folded.size()) -
```

Comment:
What does sz mean?

Reply:
It was mean to mean `size` since this number denotes the size of the following `commits` and `data`, but yes, it does sound nondescript.  I'll change it to `total_folds` since this value was supposed to represent the total number of times the while loop after runs!